Source/JavaScriptCore/runtime/LiteralParser.cpp

```diff
@@ -64,6 +64,8 @@ inline const CharType* LiteralParser<CharType, reviverMode>::Lexer::currentToken
 template<typename CharType, JSONReviverMode reviverMode>
 bool LiteralParser<CharType, reviverMode>::tryJSONPParse(Vector<JSONPData>& results, bool needsFullSourceInfo)
 {
+    ASSERT(m_mode == JSONP);
+    ASSERT(reviverMode == JSONReviverMode::Disabled);
```

Comment:
Since `JSONReviverMode` is a constant template argument, let's make this a `static_assert` instead.

Reply:
This does not work since reviverMode `Enabled` case is also instantiated.

Reply:
This assert seems redundant with the above `requires` clause.